plugins/regex_revalidate/regex_revalidate.c

```diff
@@ -115,21 +118,23 @@ free_invalidate_t(invalidate_t *i)
 }
 
 static void
-free_invalidate_t_list(invalidate_t *i)
+free_invalidate_t_list(invalidate_t *iptr)
 {
-  if (i->next) {
-    free_invalidate_t_list(i->next);
+  while (NULL != iptr) {
+    invalidate_t *const next = iptr->next;
+    free_invalidate_t(iptr);
+    iptr = next;
```

Comment:
Looks like this data structure could be an std::list.

Reply:
perhaps.  Although std::list, linked lists are the worst of the worst for performance.  std::vector pretty much wins out every time, mostly due to memory fragmentation and modern cpu caches.